### DISCUSSION TITLE
Copilot can read my local projects .env files. This data can be displayed as suggestion to another dev in the future?

### DISCUSSION THREAD

**Discussion by leticiayanaguya** [community/community#13334] [Copilot Conversations]
This can be a silly question, but I didn't found much enlightenment reading the docs.

As expected, Copilot is primarily suggesting codes related to my own repo.
I noticed that it also completes with values ​​that are inside my development environment .env files.

What is being shown to me, from my local repo, will probably be used to improve the AI suggestions. Can these kind of values eventually be visible to another developer in the future?

**Comment by stevenpollack** [comment]
Given the potentially sensitive nature of data in .env's (e.g. api keys and secrets), it'd be interesting to see a potential black-list feature where we can tell co-pilot to ignore particular paths/files...

**Comment by WarrenMfg** [comment]
Same concern. For the time being I added the following to my `settings.json` file in VS Code. Copilot no longer makes suggestions in those files, but I have no idea if it's still reading and transmitting them.

```jsonc

"github.copilot.enable": {
  "*": true,
  "plaintext": false, // for `.env` files
  "json": false, // for `local.settings.json` files
  }

```

**Comment by kwolniak** [comment]
According to this: https://twitter.com/PrimaryObjects/status/1414542721075847170 `.env` files are ignored.

**Reply by steebchen** [reply]
Not really sure if that's universally true; for example in IntelliJ Copilot works in .env files (it suggests things, so I'm assuming it also sends the contents to their servers).

**Reply by haplorhine** [reply]
This isn't the case anymore. In VS Code, copilot can read and edit `.env` files.

**Comment by daniel-krastev** [comment]
I find it very baffling that ignoring of files is allowed in the business version and not in the still paid subscription version for individuals? Currently if I open an .env files I can see copilot generating suggestions.

**Comment by ricklove** [comment]
Github copilot is working in my .env file - This seems like a security issue that is not resolved.

Note to self: don't put important stuff in .env files.

**Reply by vanyabrucker** [reply]
not being apple to put important stuff into .env files sounds kinda insane lol we need a .env.env now :D

**Comment by MohammadKurjieh** [comment]
I also noticed that the copilot chat is also referring to my `.env` when in chat mode. 

<img width="393" alt="image" src="https://github.com/user-attachments/assets/8218e2c5-d478-43b8-8b55-45b6d3e232f2" />

I also second adding the option to blacklist files, and making `.env`s as an opt-in option. I belief most developers/companies are not comfortable with sharing their `.env`'s with copilot.

**Reply by ValentinKaisermayer** [reply]
This is a big no-no

**Comment by nalladev** [comment]
Any final solution?
I'm getting copilot suggestions in .env files.

**Comment by Zalmez** [comment]
The only way I have been able to stop suggestions in the .env file is to set these in vs code settings:
```json
    "files.associations": {
        ".env*": "dotenv"
    },
    "github.copilot.enable": {
        "*": true,
        "dotenv": false,
        "plaintext": false
    }
```
But even then it doesn't really ignore the .env file if you ask copilot about anything remotely considered env

**Comment by GamearooCodes** [comment]
i fill like it could use the .gitignore most of us have to ignore files we dont want it to read or write to idk if it would be hard to be like if in .gitignore dont open and read it?

**Reply by ValentinKaisermayer** [reply]
Windsurf has `.codeiumignore` https://docs.windsurf.com/context-awareness/local-indexing#windsurfignore

**Comment by codayon** [comment]
Copilot giving me suggestions for tab completion on .env file...

**Comment by jasonjames2** [comment]
Good question about local .env privacy clarity on how this data is handled would definitely help developers feel more secure; you can also [view the available subscriptions](https://www.iptvuk.buzz/).
